In [18]:
#Import pandas
import pandas as pd


In [19]:
#Open the csv
purchases = "Resources/purchase_data.csv"

#read with pandas
purchases_df = pd.read_csv(purchases)

#display first 5 rows
purchases_df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [20]:
#Open the csv
purchases = "Resources/purchase_data.csv"

#read with pandas
purchases_df = pd.read_csv(purchases)

#display first five rows
purchases_df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [21]:
#Purchasing Analysis: Total

#Find total number of players
total_players = purchases_df["SN"].nunique()
total_players

#Number of unique items
number_items = purchases_df["Item ID"].nunique()
number_items

#Average purchase price
average_price = round(purchases_df["Price"].mean(),2)
average_price

#Total number of purchases
total_purchases = purchases_df["Item ID"].count()

#Total revenue
total_value = purchases_df["Price"].sum()
total_value

#stick it all into a dataframe

purchasing_analysis_df = pd.DataFrame({"Total Players" : total_players, 
                                       "Number of Unique Items" : number_items,
                                       "Average Purchase Price" : average_price,
                                       "Total Number of Purchases" : total_purchases,
                                       "Total Revenue" : total_value}, index=["Summary Statistics"])
#map better formats onto columns
purchasing_analysis_df["Average Purchase Price"] = purchasing_analysis_df["Average Purchase Price"].map('${:.2f}'.format)
purchasing_analysis_df["Total Revenue"] = purchasing_analysis_df["Total Revenue"].map('${:.2f}'.format)

#display
purchasing_analysis_df

Total Players  Number of Unique Items  \
Summary Statistics            576                     183   

                   Average Purchase Price  Total Number of Purchases  \
Summary Statistics                  $3.05                        780   

                   Total Revenue  
Summary Statistics      $2379.77

In [22]:
#Gender Demographics: Unique screennames
#Percentage and Count of Players, male, female, and otherwise


male_players = purchases_df.loc[purchases_df["Gender"]== "Male", :]
count_male = male_players["SN"].nunique()

female_players = purchases_df.loc[purchases_df["Gender"]=="Female", :]
count_female = female_players["SN"].nunique()

nb_players = purchases_df.loc[purchases_df["Gender"]=="Other / Non-Disclosed", :]
count_nb = nb_players["SN"].nunique()


percent_male = count_male / total_players
percent_female = count_female / total_players
percent_nb = count_nb / total_players


#Put it in a dataframe

gender_demo_df = pd.DataFrame({"Count" : [count_male, count_female, count_nb], 
                                   "Percent" : [percent_male, percent_female, percent_nb]}, index = ["Male", "Female", "Other / Non-Disclosed"]) 

#Format percent column
gender_demo_df["Percent"] = gender_demo_df["Percent"].map("{:.1%}".format)

#Display
gender_demo_df


Count Percent
Male                     484   84.0%
Female                    81   14.1%
Other / Non-Disclosed     11    1.9%

In [23]:
#Purchasing Analysis (Gender):
#Purchase Count from above
male_purchases = male_players["Price"].count()
female_purchases = female_players["Price"].count()
nb_purchases = nb_players["Price"].count()

#Average purchase price
male_avg_price = male_players["Price"].mean()
female_avg_price = female_players["Price"].mean()
nb_avg_price = nb_players["Price"].mean()

#Total purchase value
male_total_price = male_players["Price"].sum()
female_total_price = female_players["Price"].sum()
nb_total_price = nb_players["Price"].sum()

#Average purchase total per person
avg_per_person_male = male_total_price/count_male
avg_per_person_female = female_total_price/count_female
avg_per_person_nb = nb_total_price/count_nb

gender_analysis_df = pd.DataFrame({"Purchase Count" : [male_purchases, female_purchases, nb_purchases],
                                  "Average Purchase Price" : [male_avg_price, female_avg_price, nb_avg_price],
                                  "Total Purchase Value" : [male_total_price, female_total_price, nb_total_price],
                                  "Average Purchase Total Per Person" : [avg_per_person_male, avg_per_person_female, avg_per_person_nb]},
                                  index = ["Male", "Female", "Other / Non-Disclosed"])

#map dollar formats
gender_analysis_df["Average Purchase Price"] = gender_analysis_df["Average Purchase Price"].map('${:.2f}'.format)
gender_analysis_df["Total Purchase Value"] = gender_analysis_df["Total Purchase Value"].map('${:.2f}'.format)
gender_analysis_df["Average Purchase Total Per Person"] = gender_analysis_df["Average Purchase Total Per Person"].map('${:.2f}'.format)

#Display
gender_analysis_df


Purchase Count Average Purchase Price  \
Male                              652                  $3.02   
Female                            113                  $3.20   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Average Purchase Total Per Person  
Male                              $1967.64                             $4.07  
Female                             $361.94                             $4.47  
Other / Non-Disclosed               $50.19                             $4.56

In [46]:
#create a dataframe for unique SN's
sn_unique_df = purchases_df.groupby("SN").mean()
sn_unique_df.head()


Purchase ID   Age  Item ID  Price
SN                                             
Adairialis76        467.0  16.0    123.0   2.28
Adastirin33         142.0  35.0    175.0   4.48
Aeda94              388.0  17.0    128.0   4.91
Aela59               28.0  21.0    119.0   4.32
Aelaria33           630.0  23.0    171.0   1.79

In [48]:
#Create bins and bin labels
bins = [0, 10, 14, 19, 24, 29, 34, 39, 100]
bin_labels = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]

#Cut and add an age demo column to dataframe
sn_unique_df["Age Demographic"] = pd.cut(sn_unique_df["Age"], bins, labels = bin_labels)

#create a groupby to summarize
age_demo = sn_unique_df.groupby("Age Demographic").count()

#Convert to DataFrame
age_demo_df = pd.DataFrame(age_demo)

#Rename column to count
age_demo_df = age_demo_df.rename(columns = {"Age" : "Count"})
age_demo_df.reset_index()

#Calculate percentages
percentage = (age_demo_df["Count"]/total_players)

#Append percentages to DF as column
age_demo_df["Percent"] = percentage

#Mapping for format
age_demo_df["Percent"] = age_demo_df["Percent"].map("{:.1%}".format)

#Only display Count and Percentage column
age_demo_df[["Count", "Percent" ]]



Count Percent
Age Demographic               
<10                 24    4.2%
10-14               15    2.6%
15-19              107   18.6%
20-24              258   44.8%
25-29               77   13.4%
30-34               52    9.0%
35-39               31    5.4%
40+                 12    2.1%

In [26]:
#Top spenders
spenders_gb = purchases_df.groupby(["SN"])

#Create a dataframe (using count merges multiple entries)
top_spenders_df = pd.DataFrame({"Purchase Count" : spenders_gb["Purchase ID"].count(), "Average Purchase Price" : spenders_gb["Price"].mean(), "Total Purchase Value" : spenders_gb["Price"].sum()} )

#Sort dataframe descending
top_spenders_df = top_spenders_df.sort_values("Total Purchase Value", ascending = False)

#Mapping for format
top_spenders_df["Average Purchase Price"] = top_spenders_df["Average Purchase Price"].map('${:.2f}'.format)
top_spenders_df["Total Purchase Value"] = top_spenders_df["Total Purchase Value"].map('${:.2f}'.format)

#Display
top_spenders_df.head()

Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

In [27]:
#Most popular and profitable items
#Groupby items
item_gb = purchases_df.groupby(["Item ID", "Item Name", "Price"])

#Create a dataframe to include item name, count, price, and total purchase value
item_df = pd.DataFrame({"Purchase Count" : item_gb["Item ID"].count(), 
                        "Total Purchase Value" : item_gb["Price"].sum(),
                         })



In [28]:
#Most popular items:
popular_df = item_df.sort_values("Purchase Count", ascending = False)

#Map column 
popular_df["Total Purchase Value"] = popular_df["Total Purchase Value"].map('${:.2f}'.format)

#Display
popular_df.head()

Purchase Count  \
Item ID Item Name                                    Price                   
178     Oathbreaker, Last Hope of the Breaking Storm 4.23               12   
145     Fiery Glass Crusader                         4.58                9   
108     Extraction, Quickblade Of Trembling Hands    3.53                9   
82      Nirvana                                      4.90                9   
19      Pursuit, Cudgel of Necromancy                1.02                8   

                                                           Total Purchase Value  
Item ID Item Name                                    Price                       
178     Oathbreaker, Last Hope of the Breaking Storm 4.23                $50.76  
145     Fiery Glass Crusader                         4.58                $41.22  
108     Extraction, Quickblade Of Trembling Hands    3.53                $31.77  
82      Nirvana                                      4.90                $44.10  
19      Pursuit, Cudgel of Necromancy                1.02                 $8.16

In [29]:
#Most profitable items:
#Sort dataframe on total purchase value
profitable_df = item_df.sort_values("Total Purchase Value", ascending = False)

#Map column
profitable_df["Total Purchase Value"] = profitable_df["Total Purchase Value"].map('${:.2f}'.format)


#Display
profitable_df.head()

Purchase Count  \
Item ID Item Name                                    Price                   
178     Oathbreaker, Last Hope of the Breaking Storm 4.23               12   
82      Nirvana                                      4.90                9   
145     Fiery Glass Crusader                         4.58                9   
92      Final Critic                                 4.88                8   
103     Singed Scalpel                               4.35                8   

                                                           Total Purchase Value  
Item ID Item Name                                    Price                       
178     Oathbreaker, Last Hope of the Breaking Storm 4.23                $50.76  
82      Nirvana                                      4.90                $44.10  
145     Fiery Glass Crusader                         4.58                $41.22  
92      Final Critic                                 4.88                $39.04  
103     Singed Scalpel                               4.35                $34.80

Analysis

From the data in our sample, it’s clear that male players outnumber female players and players with an undisclosed identity 84% to 16%. Similarly, male players produce about 83% of total revenue for the game, though it appears that female and undisclosed players spend on average 40-50 cents more per person individually.

It’s also evident that players are, by and large, highly concentrated among teenagers and young adults, with approximately 60% of players between the ages of 14 and 22. Only 7% of players are younger than 14 and only 16% of players are older than 30.

In general, it appears that an item’s overall profitability goes up with the price of the item; the five most profitable items all have an individual price of 4 dollars or more, whereas the least profitable items sell for 2 dollars or less. There is a much less clear correlation between item popularity and item price; the five most popular items sold in the game range in price from 1.02 to 4.90. The item “Oathbreaker, Last Hope of the Breaking Storm” is both the most popular and most profitable in the game. 
